packages/database/src/migrations/20240902224836-argon2-passwords-modifies-data.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+import { hash } from '@node-rs/argon2';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.runSql(`
+    ALTER TABLE user_account
+    RENAME COLUMN password_hash TO password_hash_old;
+    
+    ALTER TABLE user_account
+    ADD COLUMN password_hash TEXT;
+  `);
+  const users = await db.runSql('SELECT id, password_hash_old, password_salt FROM user_account');
+
+  await Promise.all(
```

Comment:
There are about 4,400 user_account records in the database so this set of updates take 5 - 6 minutes (on my local) . I'm not sure if there is a precedent for long migrations like this or if it is a problem?!

Reply:
I had a chat about this with @rohan-bes and we agreed that it's fine. Our rule of thumb is
- Sub 10 mins is totally fine
- 10+ worth having a discussion with the team but generally safe
- 1 hour + probably prefer to do OTS during scheduled downtime